src/components/edit-asset-address-form/editable-address/utils.ts

```diff
@@ -73,3 +74,25 @@ export const buildAssetAddress = (
   postPreamble: editAssetAddressRequest.assetAddress.postPreamble,
   uprn: assetDetails.assetAddress.uprn,
 });
+
+export const getLlpgAddressFormValues = (editableAddress: Address) => {
+  return {
+    postPreamble: "",
+    addressLine1: editableAddress?.line1 ? editableAddress.line1 : "",
+    addressLine2: editableAddress?.line2 ? editableAddress.line2 : "",
+    addressLine3: editableAddress?.line3 ? editableAddress.line3 : "",
+    addressLine4: editableAddress?.town ? editableAddress.town : "",
+    postcode: editableAddress?.postcode ? editableAddress.postcode : "",
+  };
+};
+
+export const getCurrentAddressFormValues = (editableAddress: AssetAddress) => {
+  return {
+    postPreamble: editableAddress?.postPreamble ? editableAddress?.postPreamble : "",
+    addressLine1: editableAddress?.addressLine1 ? editableAddress.addressLine1 : "",
+    addressLine2: editableAddress?.addressLine2 ? editableAddress.addressLine2 : "",
+    addressLine3: editableAddress?.addressLine3 ? editableAddress.addressLine3 : "",
+    addressLine4: editableAddress?.addressLine4 ? editableAddress.addressLine4 : "",
+    postcode: editableAddress?.postCode ? editableAddress.postCode : "",
+  };
+};
```

Comment:
Dont both of these methods have the same functionality?

Reply:
Similar, I tried to see if typeOf could be used but in this case it couldn't, as it only works with JS types (strings, number, etc) and not custom ones/interfaces.

The difference is that they take two different parameters. One takes in an Address (which would be an LLPG suggestion) and one an AssetAddress (from our DynamoDb). The property names are a little different

Reply:
Could you use an or operator? for example `(editableAddress: AssetAddress | Address) =>`

Reply:
Alternatively, make the method more generic, and convert the types into that generic type

Reply:
Initially I tried to use (editableAddress: AssetAddress | Address) and I was trying to couple it with a typeOf to determine the type of the input parameter but I spent some time on it and got nowhere.

Only thing I can think of is to create a new type of object that the "unified" method will take, and then create a new instance of this object using the Address object and AssetAddress object. This way we have one method, one input type, but I just think it's unnecessary complexity.

Did you have anything in mind in particular? I'm out of ideas

Reply:
Changed to

```
export const getLlpgAddressFormValues = (llpgAddress: Address) => {
  return {
    postPreamble: "",
    addressLine1: llpgAddress?.line1 || "",
    addressLine2: llpgAddress?.line2 || "",
    addressLine3: llpgAddress?.line3 || "",
    addressLine4: llpgAddress?.town || "",
    postcode: llpgAddress?.postcode || "",
  };
};

export const getCurrentAddressFormValues = (currentAddress: AssetAddress) => {
  return {
    postPreamble: currentAddress?.postPreamble || "",
    addressLine1: currentAddress?.addressLine1 || "",
    addressLine2: currentAddress?.addressLine2 || "",
    addressLine3: currentAddress?.addressLine3 || "",
    addressLine4: currentAddress?.addressLine4 || "",
    postcode: currentAddress?.postCode || "",
  };
};
```

as discussed